api/src/test/java/org/openmrs/module/registrationapp/converter/RegistrationSummaryExtensionsGeneratorTest.java

```diff
@@ -0,0 +1,94 @@
+
+	
+    @Test
+    public void generate_shouldGenerateRegSummaryFromRegApp() throws Exception {
+    	
+    	InputStream inputStream = getClass().getClassLoader().getResourceAsStream("registration_app.json");    	
+    	List<AppDescriptor> appDescriptors = new ObjectMapper().readValue(inputStream, new TypeReference<List<AppDescriptor>>() {});
+    	
+    	AppDescriptor appDescriptor = appDescriptors.get(0);
+    	List<Extension> extensions = RegistrationSummaryExtensionsGenerator.generate(appDescriptor);
+    	JsonNode sections = appDescriptor.getConfig().get("sections");
+    	
+    	assertNotNull(extensions);
+    	assertEquals(3, extensions.size());
+    	for (Extension extn : extensions) {
+    		//For each Extension, loop over all appDescriptor's sections
+	    	for (JsonNode section : sections) {
+	    		String sectionId = section.get("id").getTextValue();
+	    		
+	    		if (extn.getId().contains(sectionId)) {
+					assertNotNull(extn.getId());
+					assertEquals("acme.registrationapp.summary." + sectionId, extn.getId());
+					
+					assertNotNull(extn.getAppId());
+					assertEquals(appDescriptor.getId(), extn.getAppId());
+					
+					assertNotNull(extn.getExtensionPointId());
+					assertEquals("registrationSummary.contentFragments", extn.getExtensionPointId());
+					
+					assertNotNull(extn.getExtensionParams());
+					assertEquals("summary/section", extn.getExtensionParams().get("fragment"));
+					assertEquals("registrationapp", extn.getExtensionParams().get("provider"));
+					
+					Map<String, String> fragmentConfig = (HashMap<String, String>) extn.getExtensionParams().get("fragmentConfig");
+					assertEquals(sectionId, fragmentConfig.get("sectionId"));
+					break;
+				}
+			}
+    	}
+    }
+    
+    @Test(expected = IllegalArgumentException.class)
+    public void generate_shouldThrowWhenNotRegAppConfig() throws Exception {
+    	InputStream inputStream = getClass().getClassLoader().getResourceAsStream("registration_app.json");    	
+    	List<AppDescriptor> appDescriptors = new ObjectMapper().readValue(inputStream, new TypeReference<List<AppDescriptor>>() {});
+    	
+    	AppDescriptor appDescriptor = appDescriptors.get(0);
+    	appDescriptor.setId("acme.sysadmin.manage");
+    	List<Extension> extensions = RegistrationSummaryExtensionsGenerator.generate(appDescriptor);
+    }
+    
+    @Test
+    public void generate_shouldRetrunEmptyWhenEmptySectionsInConfig() throws Exception {
+    	InputStream inputStream = getClass().getClassLoader().getResourceAsStream("registration_app.json");    	
+    	List<AppDescriptor> appDescriptors = new ObjectMapper().readValue(inputStream, new TypeReference<List<AppDescriptor>>() {});
+    	
+    	AppDescriptor appDescriptor = appDescriptors.get(0);
+    	appDescriptor.setConfig(appDescriptor.getConfig().putObject("section"));
+    	List<Extension> extensions = RegistrationSummaryExtensionsGenerator.generate(appDescriptor);
+    	assertEquals(true, extensions.isEmpty());
+    }
+}
```

Comment:
Similar idea, and fixing the typo in the method name:
```java
import static org.hamcrest.Matchers.empty;
import static org.hamcrest.core.Is.is;

@Test
public void generate_shouldReturnEmptyExtensionsWhenEmptySections() throws Exception {
  // setup
  AppDescriptor appDesc = new AppDescriptor("my.registrationapp.registerPatient", "Create a new Patient Record", "Register Patient", null, null, null, 0, null, null);

  ObjectNode config = new ObjectMapper().createObjectNode();
  config.putArray("sections"); // a config with empty sections
  appDesc.setConfig(config);

  // replay
  List<Extension> regSummaryExtensions = RegistrationSummaryExtensionsGenerator.generate(appDesc);

  // verif
  Assert.assertThat(regSummaryExtensions, is(empty()));
}
```
Also using Hamcrest matchers for readability but that's optional.

Reply:
Cool nuances @mks-d , Thank you very much. I have made changes.